Samples/SimpleIslandApp/SimpleIslandApp.cpp

```diff
@@ -0,0 +1,224 @@
+   {
+      return FALSE;
+   }
+
+   ShowWindow(hWnd, nCmdShow);
+   UpdateWindow(hWnd);
+
+   return TRUE;
+}
+
+//
+//  FUNCTION: WndProc(HWND, UINT, WPARAM, LPARAM)
+//
+//  PURPOSE: Processes messages for the main window.
+//
+//  WM_COMMAND  - process the application menu
+//  WM_PAINT    - Paint the main window
+//  WM_DESTROY  - post a quit message and return
+//
+//
+LRESULT CALLBACK WndProc(HWND hWnd, UINT message, WPARAM wParam, LPARAM lParam)
+{
+    switch (message)
+    {
+    case WM_CREATE:
+        {
+            g_button1 = ::CreateWindow(L"BUTTON", L"Win32 Button 1", WS_TABSTOP | WS_VISIBLE | WS_CHILD, 10, 10, 150, 40, hWnd, (HMENU)101, hInst, NULL);
+            g_button2 = ::CreateWindow(L"BUTTON", L"Win32 Button 2", WS_TABSTOP | WS_VISIBLE | WS_CHILD, 10, 400, 150, 40, hWnd, (HMENU)102, hInst, NULL);
+
+            // Create our DesktopWindowXamlSource and attach it to our hwnd.  This is our "island".
+            g_desktopWindowXamlSource = winrt::DesktopWindowXamlSource{};
+            g_desktopWindowXamlSource.Initialize(winrt::GetWindowIdFromWindow(hWnd));
+
+            // Put a new instance of our Xaml "MainPage" into our island.  This is our UI content.
+            g_desktopWindowXamlSource.Content(winrt::make<winrt::SimpleIslandApp::implementation::MainPage>());
+        }
+        break;
+    case WM_SIZE:
+        {
+            const int width = LOWORD(lParam);
+            const int height = HIWORD(lParam);
+
+            ::SetWindowPos(g_button1, NULL, 10, 10, 150, 40, SWP_NOZORDER);
+            ::SetWindowPos(g_button2, NULL, 10, height - 50, 150, 40, SWP_NOZORDER);
+
+            if (g_desktopWindowXamlSource)
+            {
+                g_desktopWindowXamlSource.SiteBridge().MoveAndResize({ 10, 60, width - 20, height - 120 });
+            }
+        }
+        break;
+
+    case WM_COMMAND:
+        {
+            int wmId = LOWORD(wParam);
+            // Parse the menu selections:
+            switch (wmId)
+            {
+            case IDM_ABOUT:
+                DialogBox(hInst, MAKEINTRESOURCE(IDD_ABOUTBOX), hWnd, About);
+                break;
+            case IDM_EXIT:
+                DestroyWindow(hWnd);
+                break;
+            default:
+                return DefWindowProc(hWnd, message, wParam, lParam);
+            }
+        }
+        break;
+    case WM_PAINT:
+        {
+            PAINTSTRUCT ps;
+            HDC hdc = BeginPaint(hWnd, &ps);
+            // TODO: Add any drawing code that uses hdc here...
+            EndPaint(hWnd, &ps);
+        }
+        break;
+    case WM_DESTROY:
+        PostQuitMessage(0);
+        break;
+    default:
+        return DefWindowProc(hWnd, message, wParam, lParam);
+    }
+    return 0;
+}
+
+// Message handler for about box.
+INT_PTR CALLBACK About(HWND hDlg, UINT message, WPARAM wParam, LPARAM lParam)
+{
+    UNREFERENCED_PARAMETER(lParam);
+    switch (message)
+    {
+    case WM_INITDIALOG:
+        return (INT_PTR)TRUE;
```

Comment:
should we use IntToPtr?

Reply:
I tried, the compiler still didn't want to convert void* to INT_PTR automatically.